contacts/main.js

```diff
@@ -0,0 +1,563 @@
+    this.id = person.id;
+    this.firstName = person.firstName;
+    this.lastName = person.lastName;
+    this.image = person.image;
+    this.likes = person.likes;
+  }
+  getFullNameAndLikes() {
+    return `${this.id} ${this.firstName} ${this.lastName} ${this.likes}`;
+  }
+}
+/////////////////////////////////////////////////////////////////////////
+function getInitialListOfContacts() {
+  let array = [];
+  var id;
+  var obj;
+
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Alin',
+    lastName: 'Ionescu',
+    image: 'images/profile-pics/pic1.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Leona',
+    lastName: 'Sherman',
+    image: 'images/profile-pics/pic2.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Annette',
+    lastName: 'Carlson',
+    image: 'images/profile-pics/pic3.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Ervin',
+    lastName: 'Watkins',
+    image: 'images/profile-pics/pic4.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Philip',
+    lastName: 'Pope',
+    image: 'images/profile-pics/pic5.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Viorel',
+    lastName: 'Popescu',
+    image: 'images/profile-pics/pic6.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  id = generateUniqueID();
+  obj = {
+    id: id,
+    firstName: 'Elon',
+    lastName: 'Musk',
+    image: 'images/profile-pics/pic7.jfif',
+    likes: 0
+  };
+  array.push(obj);
+  let retunrArray = [];
+  array.forEach((item) => {
+    let person = new Person(item);
+    retunrArray.push(person);
+  })
+  return retunrArray;
+}
+
+function saveDataToLocalStorage() {
+  localStorage.setItem('contacts', JSON.stringify(contactsList));
+}
+
+function getContactsFromLocalStorage() {
+  return JSON.parse(localStorage.getItem('contacts'));
+}
+
+function getContactsAccordingly() {
+  let array = getContactsFromLocalStorage();
+  // if we have nothing in the local storage we get the initial list
+  if (array === null || array.length === 0)
+    array = getInitialListOfContacts()
+  return array;
+}
+
+/////////////////////////////////////////////////////////////////////////
+
+function getNumberOfNonZeroLikes() {
+  let nonZeros = 0;
+  contactsList.forEach((person) => {
+    if (person.likes !== 0)
+      nonZeros += 1;
+  })
+  return nonZeros;
+}
+
+function refreshFavsList() {
+  // calculate the new list 
+  let favsMap = calculateFavsList();
+  // clear
+  clearFavsListHTML();
+  // rerender the new list if there is at least one nonzero elem
+  if (getNumberOfNonZeroLikes() !== 0)
+    rerenderFavsListHTML(favsMap);
+}
+
+function calculateFavsList() {
+  // returns a map
+  let maap = {};
```

Comment:
There is no need to declare the variable and set it to empty object to be used later on. You can remove this lane and on lane 139 just have something like:

const maap = createMapOfFavs(arr);